src/sub/payments.py

```diff
@@ -537,33 +510,46 @@ def create_update_data(customer) -> Dict[str, Any]:
         except KeyError:
             product = retrieve_stripe_product(subscription["plan"]["product"])
             products[subscription["plan"]["product"]] = product
-        plan_name = format_plan_nickname(
-            product_name=product["name"], plan_interval=subscription["plan"]["interval"]
-        )
+
         if subscription.get("status", None) == "incomplete":
             invoice = retrieve_stripe_invoice(subscription["latest_invoice"])
             if invoice["charge"]:
                 intents = retrieve_stripe_charge(invoice["charge"])
                 intents = intents.to_dict()
                 return_data["subscriptions"].append(
-                    {
-                        "current_period_end": subscription["current_period_end"],
-                        "current_period_start": subscription["current_period_start"],
-                        "ended_at": subscription["ended_at"],
-                        "plan_name": plan_name,
-                        "plan_id": subscription["plan"]["id"],
-                        "status": subscription["status"],
-                        "cancel_at_period_end": subscription["cancel_at_period_end"],
-                        "subscription_id": subscription["id"],
-                        "failure_code": intents["failure_code"],
-                        "failure_message": intents["failure_message"],
-                    }
+                    format_subscription(subscription, product, intents)
                 )
                 continue
```

Comment:
same as above, continue is not required and can have unexpected effects.

Reply:
updated